client/src/pages/Profile.js

```diff
@@ -1,37 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Route, Switch } from 'react-router-dom';
 import { makeStyles } from '@material-ui/core/styles';
-import { Grid } from '@material-ui/core';
+import { Grid, LinearProgress } from '@material-ui/core';
+import axios from 'axios';
 
 // Custom component imports
 import UserInfo from '../components/Profile/UserInfo';
-import Courses from '../components/Profile/Courses';
 import Settings from '../components/Profile/Settings';
 import Notifications from '../components/Profile/Notifications';
-import Drawer from '../components/Profile/Sidebar';
+import Sidebar from '../components/Profile/Sidebar';
+import Navbar from '../pages/layout/Navbar';
+import MyCourses from '../pages/dashboard/MyCourses';
 
-const useStyles = makeStyles({
+const useStyles = makeStyles(theme => ({
     container: {
-        display: 'flex',
+        height: 'calc(100vh - 100px)',
+        marginTop: 3,
     },
-});
+    contentContainer: {
+        backgroundColor: theme.palette.common.white,
+    },
+}));
 
-const Profile = props => {
+const Profile = () => {
     const classes = useStyles();
 
+    const [isLoading, setIsLoading] = useState(true);
+    const [university, setUniversity] = useState('');
+    const [userCourses, setUserCourses] = useState([]);
+    const [uniCourses, setUniCourses] = useState([]);
+
```

Comment:
I'm a little confused by the purpose of Dashboard.js and NavBar now. I don't think Dashboard is used anymore. Is profile now meant to be the higher level component? From my understanding the user information needs to be accessible from many pages/components in the app but we get it here each time we load the Profile page.

Reply:
Yes, I added a description stating that Profile has replaced Dashboard but forgot to add it in my PR. So Dashboard will be deleted in the next commit. 
For now, I am storing user info about university and courses in the Profile component level and passing props to child component (i.e. My Courses ) but at some point in the future it will be moved to the context store and it will be accessible from all pages.

Reply:
I have removed Dashboard.js and moved MyCourses.js to Profile file.

Reply:
Just curious, what was the reason for replacing Dashboard? There are a few things I'm wondering about having the Profile page as the higher level component.

Reply:
No special reason. Both of them are the skeletons that wrap the Navbar, Sidebar, and right side contents.
It's just Jon was already working on it, so we moved the Navbar and Sidebar there. 
But regarding the Navbar, I see your point it can be moved to a higher-level component once we start working on the chat, form, and groups page.

Reply:
Okay sounds good. I think we need some higher-level component because, for instance, right now when we click on the Forum tab and then go back to the profile page it makes a request to the backend to get the user again and we see the loading bar. It might look a bit weird for the demo when @ethanrosenberg group page is merged and we navigate between these two routes. It'd be good if the higher level component just handled this once on the initial load.

Reply:
i agree and think we don't want to make another request

Reply:
One of the items in the ticket I have is building out a userContext file so that we can load it from the Profile component.

Reply:
@jonquag yeah so just to clarify a bit, while the userContext will allow all the pages/nested components to get the user information, the main consideration here is in what component we're fetching the user information from. In the current setup we're doing it in the Profile page. But the problem is when we navigate to a different route this component un-mounts and then gets re-mounted when the user navigates back to the Profile page. This results in the componentDidMount function (line 47 in Profile.js) getting called again and another request to get the user information is sent (and only the loading bar is shown during this process). Ideally we'd only want to make one request to get the users information from a higher-level component that renders once when the app loads and there in sets the initial userContext; resulting in one request and one blank screen with the loading bar. Anybody please correct me if I got anything wrong

Reply:
If that’s the case that would be the first place we are routing to after sign in. With the layout component @amha19 had set up before we combined it with profile might work.

Reply:
I have created a wrapper component that wraps all the pages that need the Navbar. So, now the user school and course info are being requested in that higher-level component so a logged-in user can browse between those pages without making a request again and again.

const Layout = ({ children }) => {
    const { dispatch } = useGlobalContext();

    useEffect(() => {
        actions.fetchProfile()(dispatch);
    }, [dispatch]);

    return (
        <>
            <Navbar />
            {children}
        </>
    );
};

App.js
<Layout>
                 <Route exact path="/profile" component={Profile} />
                 <Route
                      path="/profile/:profileContent"
                                    component={Profile}
                     />
              <ProtectedRoute path="/forum" component={Forum} />
              <ProtectedRoute path="/groups" component={Groups} />
              <ProtectedRoute path="/chat" component={Chat} />
     </Layout>